containerd-shim-v2/utils.go

```diff
@@ -36,53 +35,17 @@ func cReap(s *service, status int, id, execid string, exitat time.Time) {
 func cleanupContainer(ctx context.Context, sid, cid, bundlePath string) error {
 	logrus.WithField("Service", "Cleanup").WithField("container", cid).Info("Cleanup container")
 
-	rootfs := filepath.Join(bundlePath, "rootfs")
-	sandbox, err := vci.FetchSandbox(ctx, sid)
+	err := vci.CleanupContainer(ctx, sid, cid)
 	if err != nil {
-		return err
-	}
-
-	status, err := sandbox.StatusContainer(cid)
-	if err != nil {
-		logrus.WithError(err).WithField("container", cid).Warn("failed to get container status")
-		return err
-	}
-
-	if oci.StateToOCIState(status.State.State) != oci.StateStopped {
-		err := sandbox.KillContainer(cid, syscall.SIGKILL, true)
-		if err != nil {
-			logrus.WithError(err).WithField("container", cid).Warn("failed to kill container")
-			return err
-		}
-	}
-
-	if _, err = sandbox.StopContainer(cid); err != nil {
-		logrus.WithError(err).WithField("container", cid).Warn("failed to stop container")
-		return err
+		logrus.WithError(err).WithField("container", cid).Warn("failed to cleanup container")
 	}
 
-	if _, err := sandbox.DeleteContainer(cid); err != nil {
-		logrus.WithError(err).WithField("container", cid).Warn("failed to remove container")
-	}
+	rootfs := filepath.Join(bundlePath, "rootfs")
```

Comment:
I know this isn't part of your change, but since it seems like it affects cleanup success, why are we not returning the unmount error below?

Reply:
Aha, maybe missed, I'll add it.

Reply:
/cc @amshinde